index.bs

```diff
@@ -688,50 +677,32 @@ To <dfn>compare two keys</dfn> |a| and |b|, run these steps:
 
       <dt>*string*</dt>
       <dd>
-        1. Let |length| be the lesser of |va|'s length and |vb|'s length.
-        1. Let |i| be 0.
-        1. While |i| is less than |length|, then:
-            1. Let |u| be the code unit of |va| at index |i|.
-            1. Let |v| be the code unit of |vb| at index |i|.
-            1. If |u| is greater than |v| then return 1.
-            1. If |u| is less than |v| then return -1.
-            1. Increase |i| by 1.
-        1. If |va|'s length is greater than |vb|'s length, then return 1.
-        1. If |va|'s length is less than |vb|'s length, then return -1.
+        1. If |va| is [=code unit less than=] |vb|, then return -1.
+        1. If |vb| is [=code unit less than=] |va|, then return 1.
         1. Return 0.
       </dd>
 
       <dt>*binary*</dt>
       <dd>
-        1. Let |length| be the lesser of |va|'s length and |vb|'s length.
-        1. Let |i| be 0.
-        1. While |i| is less than |length|, then:
-
-            1. Let |u| be the {{octet}} in |va| at index |i|.
-            1. Let |v| be the {{octet}} in |vb| at index |i|.
-            1. If |u| is greater than |v| then return 1.
-            1. If |u| is less than |v| then return -1.
-            1. Increase |i| by 1.
-
-        1. If |va|'s length is greater than |vb|'s length, then return 1.
-        1. If |va|'s length is less than |vb|'s length, then return -1.
+        1. If |va| is [=byte less than=] |vb|, then return -1.
+        1. If |vb| is [=byte less than=] |va|, then return 1.
         1. Return 0.
       </dd>
 
       <dt>*array*</dt>
       <dd>
-        1. Let |length| be the lesser of |va|'s length and |vb|'s length.
+        1. Let |length| be the lesser of |va|'s [=list/size=] and |vb|'s [=list/size=].
         1. Let |i| be 0.
         1. While |i| is less than |length|, then:
 
-            1. Let |u| be the [=/key=] in |va| at index |i|.
-            1. Let |v| be the [=/key=] in |vb| at index |i|.
+            1. Let |u| be the [=/key=] at |va|[|i|].
```

Comment:
At or in?

Reply:
I was thinking the temporary _u_ and _v_ variables could be removed entirely, which makes this moot. I'll do that.